docs/illink-options.md

```diff
@@ -6,14 +6,29 @@ The `illink` is IL linker version shipping with .NET Core or .NET 5 platforms. I
 the .NET SDK and most of the options are accessible using msbuild properties but any option
 can also be passed using `_ExtraTrimmerArgs` property.
 
-### Linking from the main assembly
+### Linking from the root assembly
 
 The command:
 
 `illink -a Program.exe`
 
-will use the assembly `Program.exe` as a source. That means that the linker will
-start with the main entry point method of `Program.exe` (typically the `Main` method) and process all its dependencies to determine what is necessary for this assembly to run.
+will use the assembly `Program.exe` as root linker input. That means that the linker will
+start with the main entry point method of `Program.exe` (typically the `Main` method) and
+process all its dependencies to determine what is necessary for this assembly to run.
+
+It's possible to use multiple input files and linker will treat them as multiple sources of
+input. When a library is used instead of executable linker will mark all members instead. This
+behaviour can be further customized using `--root` option which allows the following options.
+
+- `all` - Keep all members in root assembly
+- `default` - Use entry point for applications and all members for libraries
+- `entrypoint` - Use assembly entry point as only root in the assembly
+- `visible` - Keep all members and types visible outside of root assembly
+
+You can retain all public members of `Program.exe` application by calling
+linker like
+
+`illink -a Program.exe --root visible Program`
```

Comment:
The reason is that I want to keep -a simple and --root is advanced rarely used option. -a, -p and --root have different intend.

> "--root all Program" is giving us another way to say "-p copy Program"

Yes, but why is this problematic?

> I think "--root" should be combined with either "-a" or "-p".

What would be the benefit of pushing rarely used option to common options?

Reply:
Keeping "-a" simple makes sense, I just don't like the redundancy if you are using "--root". Maybe we could have "--root" imply "-a"? Or have syntax like `-a <assembly> [optional root mode]`. I personally would vote for `--root <assembly> [optional root mode]` to replace "-a".

> why is this problematic?

Just because it's another concept to keep track of. The assembly actions are already a source of confusion, and it would be best not to further muddy the concepts.

In my mind, 'copy' used to mean something different - it used to affect the output but not the rooting behavior - it would keep the whole assembly possibly with dangling references, because it wasn't necessarily marked. But now that 'copy' has changed to mean "root everything" I think it makes sense to get rid of '-p copy' and use '--root all' instead.

Reply:
Originally I didn't want to add it to `-a` because that will make it harder to remove `-a` as it's a redundant option (you want `illink myfile.dll` just work) but thinking about it again it should be possible to make it work with that for most cases.